src/master/master.py

```
@@ -18,27 +21,64 @@
 
 logger = Logger()
 
+
 class HyperMaster():
     """
-    When the master is started, this class should be what the user application
-    can import to begin using the features of the system on the master itself
+    HyperMaster Class.
     """
-
-    def __init__(self, HOST="0.0.0.0", PORT=5678):
-        self.host = HOST
-        self.port = PORT
+    jobfile_name: str = None
+    jobfile_path: str = None
+    job_path: str = None
+
+    def __init__(self, host="0.0.0.0", port=5678, jobfile_name='jobfile'):
+        self.host = host
+        self.port = port
+        self.jobfile_name = jobfile_name
         self.test_config = None
         self.task_queue = []
         self.conn_manager: ConnectionManager = ConnectionManager()
 
+    def init_job(self):
+        """
+        Initializes the job for the master.
+        Ensures jobfile is readable and job files exist
+        """
+
+        cwd = str(Path.cwd().resolve())
+        job_root_dir_path = cwd + '/job'
+        Path.mkdir(Path(job_root_dir_path), parents=True, exist_ok=True)
+
+        jobfile_path = f'{job_root_dir_path}/{self.jobfile_name}'
+
+        while not Path(jobfile_path).exists():
+            logger.log_warn(
+                f'{self.jobfile_name} not found. Please provide one in {job_root_dir_path}')
+            logger.log_info('Sleeping for 3 seconds. Zzz...')
+            sleep(3)
+
+        with open(jobfile_path, "r") as job_file:
+            job_json = json_loads(job_file.read())
+            job_id = job_json.get("job_id")
+            job_file_names: list = job_json.get("file_names")
+
+        for file_name in job_file_names:
```

Do we want this to have a different "file does not exist" behaviour than the jobfile check?

---

Yes because this is checking the validity of the jobfile provided. 

If the files the jobfile is listing don't exist, the job cannot be processed. 

We can also make  it interactive, where we tell the master to check when everything is ready, but that would be a separate issue. I figured if we were planning on making a GUI later on, then the proposed method here is perfectly viable.

---

Or we can wait for each file to be provided, with like maybe a max retry count, and then error out if not provided then?

---

Possibly, the reason I ask is because we had talked about exiting the program entirely when the jobfile is not present, instead of waiting for it to appear.

---

Oh right, 

The reason I made it this way was because the initial startup, setting up the docker and stuff, has some overhead, and it felt cleaner to have it not just end right away and wait for a job.

It's actually not even starting up the server yet so it's not even "running". It starts the server up once a job is initialized.  

It's just different behaviour that ends with the same result. If we want to exit out instead of waiting, that's fine too. Easy change

---

If we start to think about a GUI more then this may come up again. Regardless, this will be fine for now!